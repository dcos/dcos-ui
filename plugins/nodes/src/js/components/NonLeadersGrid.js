import React from "react";

import ConfigurationMap from "#SRC/js/components/ConfigurationMap";
import ConfigurationMapHeading from "#SRC/js/components/ConfigurationMapHeading";
import ConfigurationMapLabel from "#SRC/js/components/ConfigurationMapLabel";
import ConfigurationMapRow from "#SRC/js/components/ConfigurationMapRow";
import ConfigurationMapSection from "#SRC/js/components/ConfigurationMapSection";
import ConfigurationMapValue from "#SRC/js/components/ConfigurationMapValue";
import Loader from "#SRC/js/components/Loader";
import StringUtil from "../../../../../src/js/utils/StringUtil";

const Loading = () => <Loader size="small" type="ballBeat" />;

const ConfigurationRow = ({ keyValue, title, value }) => {
  return (
    <ConfigurationMapRow key={keyValue}>
      <ConfigurationMapLabel>{title}</ConfigurationMapLabel>
      <ConfigurationMapValue>
        <span className={value.classNames}>
          {StringUtil.capitalize(value.title)}
        </span>
      </ConfigurationMapValue>
    </ConfigurationMapRow>
  );
};

const NonLeader = ({ master }) => {
  return (
    <ConfigurationRow
      keyValue={master.host_ip}
      title={master.host_ip}
      value={master.healthDescription}
    />
  );
};

const EmptyLeaderList = () => {
  return <div>There are no more known masters in this cluster.</div>;
};

const NonLeaderList = ({ masters }) => {
  if (masters.length === 0) {
    return <EmptyLeaderList />;
  }

  return (
    <ConfigurationMapSection>
      {masters.map(master => (
        <NonLeader key={master.host_ip} master={master} />
      ))}
    </ConfigurationMapSection>
  );
};

export default function NonLeaderGrid({ masters }) {
  const hasMasters = Array.isArray(masters);
  const content = hasMasters ? (
    <NonLeaderList masters={masters} />
  ) : (
    <Loading />
  );

  return (
    <div className="container">
      <ConfigurationMap>
        <ConfigurationMapHeading className="flush-top">
          Non-Leaders
        </ConfigurationMapHeading>
        {content}
      </ConfigurationMap>
    </div>
  );
}
