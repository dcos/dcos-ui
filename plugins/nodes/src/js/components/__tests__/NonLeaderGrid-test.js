import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import enUS from "#SRC/js/translations/en-US.json";

import NonLeadersGrid from "../NonLeadersGrid";

const masters = [
  {
    health: 0,
    healthDescription: {
      classNames: "text-success",
      key: "HEALTHY",
      sortingValue: 3,
      title: "Healthy",
      value: 0
    },
    host_ip: "172.31.10.48",
    role: "master"
  },
  {
    health: 0,
    healthDescription: {
      classNames: "text-success",
      key: "HEALTHY",
      sortingValue: 3,
      title: "Healthy",
      value: 0
    },
    host_ip: "172.31.10.49",
    role: "master"
  }
];

const TestableNonLeadersGrid = ({ masters }) => {
  return (
    <IntlProvider locale="en" messages={enUS}>
      <NonLeadersGrid masters={masters} />
    </IntlProvider>
  );
};

describe("LeaderGrid", function() {
  it("renders loading", function() {
    expect(
      renderer.create(<TestableNonLeadersGrid masters={undefined} />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders empty", function() {
    expect(
      renderer.create(<TestableNonLeadersGrid masters={[]} />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders masters", function() {
    expect(
      renderer.create(<TestableNonLeadersGrid masters={masters} />).toJSON()
    ).toMatchSnapshot();
  });
});
