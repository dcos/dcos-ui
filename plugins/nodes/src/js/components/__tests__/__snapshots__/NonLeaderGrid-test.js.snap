// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderGrid renders empty 1`] = `
<div
  className="container"
>
  <div
    className="configuration-map"
  >
    <h1
      className="detail-view-section-heading detail-view-section-heading-primary configuration-map-heading flush-top"
    >
      Non-Leaders
    </h1>
    <div>
      There are no more known masters in this cluster.
    </div>
  </div>
</div>
`;

exports[`LeaderGrid renders loading 1`] = `
<div
  className="container"
>
  <div
    className="configuration-map"
  >
    <h1
      className="detail-view-section-heading detail-view-section-heading-primary configuration-map-heading flush-top"
    >
      Non-Leaders
    </h1>
    <div
      className="loader horizontal-center"
    >
      <div
        className="ball-beat loader--small"
      >
        <div
          className="loader-element"
        />
        <div
          className="loader-element"
        />
        <div
          className="loader-element"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`LeaderGrid renders masters 1`] = `
<div
  className="container"
>
  <div
    className="configuration-map"
  >
    <h1
      className="detail-view-section-heading detail-view-section-heading-primary configuration-map-heading flush-top"
    >
      Non-Leaders
    </h1>
    <div
      className="configuration-map-section"
    >
      <div
        className="configuration-map-row table-row"
      >
        <div
          className="configuration-map-label"
        >
          172.31.10.48
        </div>
        <div
          className="configuration-map-value"
        >
          <span
            className="text-success"
          >
            Healthy
          </span>
        </div>
      </div>
      <div
        className="configuration-map-row table-row"
      >
        <div
          className="configuration-map-label"
        >
          172.31.10.49
        </div>
        <div
          className="configuration-map-value"
        >
          <span
            className="text-success"
          >
            Healthy
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
